### PR TITLE
Made create and delete idempotent.

### LINKED ISSUE #5: Make create and delete idempotent
At the moment, running `create` while the cluster is running will result in an error (because the containers already exist):

```
$ footloose create
INFO[0000] Image: quay.io/footloose/centos7 present locally 
INFO[0000] Creating machine: cluster-node0 ...          
ERRO[0000] docker: Error response from daemon: Conflict. The container name "/cluster-node0" is already in use by container "5134f80b733ecf4561f154c7c43eca2fb5273ae8da0f7c73402b2205ceef1a3e". You have to remove (or rename) that container to be able to reuse that name. 
ERRO[0000] See 'docker run --help'.                     
```
Same with `delete`.

We should make them idempotent.

### PR DESCRIPTION
**Pull request by Skarlso**
Closes #5.

### CODE REVIEW

**Comment by Skarlso** [comment]
Going to try and add some coverage as well. :) Tomorrow though.. :)

File: tests/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitation can be lifted once we can select `footloose` containers better
 To run all tests:
 
 ```console
-go tests -v ./tests
+go test -v ./tests
```
Comment (dlespiau) [inline tests/README.md]: There are two `go tests` typos in the file, could you fix the other one while you're there :)

**Comment by dlespiau** [comment]
You should be able to run your new test locally for debugging with:

```
go test -v -run TestEndToEnd/test-create-delete-idempotent-fedora29
```

File: tests/test-create-delete-idempotent-%image.output [hunk 2]
```diff
@@ -0,0 +1,2 @@
+success
```
Comment (dlespiau) [inline tests/test-create-delete-idempotent-%image.output]: These reference output files are named `.golden.output`, the e2e tests won't look at just `.output` files.

**Comment by Skarlso** [comment]
Yeah I was looking for that. I'll do that.. Just, my daughter woke up so I'm off for now. I kind of thought I'll wing it. 😂

File: tests/test-create-delete-idempotent-%image.cmd [hunk 3]
```diff
@@ -0,0 +1,5 @@
+footloose config create --config %testName.footloose --name %testName --key %testName-key --image quay.io/footloose/%image
+footloose create --config %testName.footloose
+%out footloose create --config %testName.footloose echo success
```
Comment (dlespiau) [inline tests/test-create-delete-idempotent-%image.cmd]: `create` doesn't take any argument like `ssh` does, so `echo success` will not work here.

I think the easier way to do it is to do:

```
footloose create --config %testName.footloose
footloose create --config %testName.footloose
footloose delete --config %testName.footloose
footloose delete --config %testName.footloose
```

If one of those commands fails, the test will fail. If they all succeed, the test will succeed, which seems to the right thing when testing for idempotency.

**Comment by dlespiau** [comment]
> Yeah I was looking for that. I'll do that.. Just, my daughter woke up so I'm off for now. I kind of thought I'll wing it.

Ah, more important things have preempted you :) I left a few remarks that should help. Take your time, there's nothing urgent about this PR, it will wait for when you have some time.

**Comment by Skarlso** [comment]
Thanks! Works. :)

--- PASS: TestEndToEnd (2.58s)
    --- PASS: TestEndToEnd/test-create-delete-idempotent-fedora29 (2.58s)
PASS
ok      github.com/weaveworks/footloose/tests   2.601s

**Comment by dlespiau** [comment]
Thanks for your contribution, merging!

I think we'll want for re-factor "is the machine running or not?" on the `Machine` object at some point (https://github.com/weaveworks/footloose/blob/master/pkg/cluster/machine.go). But that can be done later when needed it in a different place.

**Comment by Skarlso** [comment]
@dlespiau yeah, agreed. :) As a nicer way of doing it, by just calling a method. Something like, `isRunning()`. I'll take a look at that later on. :)

Thank you!

**Pull request merged**